projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1927,6 +1927,10 @@ protected static function build_sprintf( $args ) {
 	 */
 	public function publicize_connections_url( $source = 'calypso-marketing-connections' ) {
 		if ( $this->use_admin_ui_v1() && current_user_can( 'manage_options' ) ) {
+			if ( Publicize_Script_Data::has_feature_flag( 'admin-page' ) ) {
+				return ( new Paths() )->admin_url( array( 'page' => 'jetpack-social' ) );
+			}
+
 			$is_social_active = defined( 'JETPACK_SOCIAL_PLUGIN_DIR' );
```

Comment:
I realise this is getting a bit nit-picky, but what about adding it to the `is_social_active` check? It might be worth changing that name. `has_social_admin` or something.

```suggestion
			$is_social_active = defined( 'JETPACK_SOCIAL_PLUGIN_DIR' ) || Publicize_Script_Data::has_feature_flag( 'admin-page' );
```

Reply:
That would already return early in the if above